fml/mapping.h

```diff
@@ -125,6 +125,44 @@ class NonOwnedMapping final : public Mapping {
   FML_DISALLOW_COPY_AND_ASSIGN(NonOwnedMapping);
 };
 
+/// A Mapping like NonOwnedMapping, but uses Free as its release proc.
+class MallocMapping final : public Mapping {
+ public:
+  MallocMapping();
+
+  MallocMapping(const uint8_t* data, size_t size);
+
+  MallocMapping(fml::MallocMapping&& mapping);
+
+  ~MallocMapping() override;
+
+  /// Copies the data from `begin` to `end`.
+  /// It's templated since void* arithemetic isn't allowed and we want support
```

Comment:
Not sure I understand this reasoning. `void *` arithmetic is disallowed but the caller can just reinterpret_cast it though right?

Reply:
Yep, they could.  I was trying to make the migration as easy as possible, just added what I made it easiest.  The problem was that we use uint8_t everywhere but when you are creating mappings from strings they use int8_t.